### PR TITLE
Use UntilWithSync() instead of untilWithoutRetry() for traces watching

### PR DESCRIPTION
**Pull request by eiffel-fl**
Hi.

In this PR, I replaced `untilWithoutRetry()` by `UntilWithSync()`.
So, I was able to delete the local copy of `untilWithoutRetry()` code and we do not need to worry about it being [made private](https://github.com/kubernetes/client-go/blob/c6bd30b9ec5f668df191bc268c6f550c37726edb/tools/watch/until.go#L55) one day.

Best regards.

### CODE REVIEW

File: cmd/kubectl-gadget/utils/trace.go [hunk 1]
```diff
@@ -507,23 +469,23 @@ func waitForCondition(traceID string, conditionFunction func(*gadgetv1alpha1.Tra
...
 		if err != nil {
 			return nil, err
 		}
 
 		ctx, cancel := watchtools.ContextWithOptionalTimeout(context.Background(), TraceTimeout)
-		_, err = untilWithoutRetry(ctx, watcher, func(event watch.Event) (bool, error) {
+		defer cancel()
+
+		createdTraces := 0
+		_, err = watchtools.UntilWithSync(ctx, traceListerWatcher, &gadgetv1alpha1.Trace{}, nil, func(event watch.Event) (bool, error) {
```
Comment (blanquicet) [inline cmd/kubectl-gadget/utils/trace.go:488]: Why we were not using `UntilWithSync` from the very beginning?

File: cmd/kubectl-gadget/utils/trace.go [hunk 2]
```diff
@@ -507,23 +469,23 @@ func waitForCondition(traceID string, conditionFunction func(*gadgetv1alpha1.Tra
...
+		// We will need to list and watch events on them.
+		// For this, we will get a ListerWatcher on all the traces which share the
+		// same ID.
+		// NOTE all the traces on different nodes but linked to one gadget share
 		// the same ID.
 		// We will also begin to monitor events since the above GET of the traces
-		// list.
-		watcher, err = getTraceWatcher(traceID, traceList.ListMeta.ResourceVersion)
+		// list thanks to resource version.
+		traceListerWatcher, err := getTraceListerWatcher(traceID, traceList.Items[tracesNumber-1].ResourceVersion)
```
Comment (blanquicet) [inline cmd/kubectl-gadget/utils/trace.go:479]: What is the difference between `traceList.ListMeta.ResourceVersion` and `traceList.Items[tracesNumber-1].ResourceVersion`?

**Review by blanquicet — COMMENTED** [review]
I'm still not sure about this PR. Could you please describe why this change in needed in the PR's description? I think I am missing some context 🙂 .

File: cmd/kubectl-gadget/utils/trace.go (hunk 1 above)
Comment (blanquicet) [reply cmd/kubectl-gadget/utils/trace.go:490]: It is clear now.

**Review by blanquicet — APPROVED** [review]
I tested it, and it is working correctly. This PR only modifies the logic for the `Added` case, which, in my opinion, is not necessary to manage.

Let's see if someone else has an opinion about it, so please don't merge it yet.

**Review by mauriciovasquezbernal — APPROVED** [review]
LGTM. I'm also not sure if handling the added event is needed, however it doesn't hurt to have this code there.

**Pull request merged**